dask_cuda/utils.py

```diff
@@ -21,3 +22,25 @@ def get_device_total_memory(index=0):
     """
     with cuda.gpus[index]:
         return cuda.current_context().get_memory_info()[1]
+
+
+def convert_frames_to_numba(frames):
+    try:
+        import cupy
+    except ImportError:
+        return frames
+
+    return [cuda.as_cuda_array(f) if isinstance(f, cupy.ndarray) else f for f in frames]
+
+
+def move_frames_to_host(frames):
+    # Conversion of frames to Numba can/should be eventually done during the
+    # serialization process in dask/distributed
+    frames = convert_frames_to_numba(frames)
+
+    return [f.copy_to_host() if isinstance(f, cuda.cudadrv.devicearray.DeviceNDArray)
+            else f for f in frames]
+
+
+def move_frames_to_device(frames):
+    return [cuda.to_device(f) if isinstance(f, np.ndarray) else f for f in frames]
```

Comment:
What are the other types of frames that might be here?

Reply:
Same as above, since not all frames are necessarily device-objects, we have to make sure not to deserialize them.